docs/lang/articles/reference.md

```diff
@@ -437,3 +437,111 @@ and it continues with the next cycle of the nearest enclosing loop.
 ### The `while` statement
 
 ### The `for` statement
+
+The `for` statement in Taichi is used to iterate over a range of numbers, multidimensional ranges, or the indices of elements in a field.
+
+```
+for_stmt        ::= "for" target_list "in" iter_expression ":" suite
+iter_expression ::= static_expression | expression
+```
+
+Taichi does not support `else` clause in `for` statements.
+
+The `for` loops can iterate in parallel if they are in the outermost scope.
+When a `for` loop is parallelized, the order of iteration is not determined,
+and it cannot be terminated by `break` statements.
+
+Taichi uses `ti.loop_config` function to set directives for the loop right after it.
+You can write `ti.loop_config(serialize=True)` before a range/ndrange `for` loop to let it run serially,
+then it can be terminated by `break` statements.
+
+There are four kinds of `for` statements:
+
+- The range `for` statement
+- The ndrange `for` iterates
+- The struct `for` statement
+- The static `for` statement
+
+#### The range `for` statement
+
+The range `for` statement iterates over a range of numbers.
+
+The `iter_expression` of range `for` statement must be like `range(begin, end)` or `range(end)`,
+and they mean the same as the python `range` function,
```

Comment:
```suggestion
and they mean the same as the Python `range` function,
```
1. Let's consistently use `Python` in this article.
2. Add a link to Python `range` so that people can know where `step` comes from.